src/sbvr-api/permissions.coffee

```diff
@@ -122,7 +123,28 @@ exports.setup = (app, sbvrUtils) ->
 		PREPARSE: ({ req }) ->
 			apiKeyMiddleware(req)
 		POSTPARSE: ({ req, request }) ->
+			if (request.method == 'POST' and request.odataQuery.property?.resource == 'canAccess')
+				request.odataQuery.options = { '$select': { 'properties': [ { 'name': 'id' } ] } }
+				{ action, method } = request.values
+				if method? and methodPermissions[method]?
+					request.permissionType = methodPermissions[method]
+				else if action?
+					request.permissionType = action
+				else
+					throw new MissingOrInvalidArgumentsError()
+				request.resourceName = request.resourceName.slice(0, request.resourceName.length - '#canAccess'.length)
+				request.odataQuery.resource = request.resourceName
+				delete request.odataQuery.property
+				request.method = 'GET'
+				request.custom.isAction = 'canAccess'
 			addPermissions(req, request)
+		PRERESPOND: ({ request, data }) ->
+			if (request.custom.isAction == 'canAccess')
+				if (_.isArray(data) and data.length == 1)
+					data[0] = _.pick(data[0], 'id', '__metadata')
```

Comment:
Why do we return the `id` rather than just true/false?

Reply:
I thought this makes more sense, in case we ever want to extend this to get all devices one can access for a specific action, in that case all returned `ids` are allowed.